sycl/source/detail/device_info.hpp

```diff
@@ -564,6 +565,27 @@ struct get_device_info_impl<id<Dimensions>,
   }
 };
 
+template <>
+struct get_device_info_impl<
+    ext::oneapi::experimental::architecture,
+    ext::oneapi::experimental::info::device::architecture> {
+  static ext::oneapi::experimental::architecture get(const DeviceImplPtr &Dev) {
+    if (!Dev->is_gpu() || backend::ext_oneapi_level_zero != Dev->getBackend())
+      throw sycl::exception(
+          make_error_code(errc::runtime),
+          "sycl_ext_oneapi_device_architecture feature is currently supported "
+          "only on GPU device with sycl::backend::ext_oneapi_level_zero or "
+          "sycl::backend::opencl backends");
```

Comment:
Could you please align this check and exception message. Currently we throw exception if we have OpenCL GPU.

Reply:
Oh yes, good catch! For debugging purposes removed OpenCL, and didn't add it back.